java/src/main/native/ai_onnxruntime_OnnxSparseTensor.c

```diff
@@ -0,0 +1,418 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * Licensed under the MIT License.
+ */
+#include <jni.h>
+#include <math.h>
+#include "onnxruntime/core/session/onnxruntime_c_api.h"
+#include "OrtJniUtil.h"
+#include "ai_onnxruntime_OnnxSparseTensor.h"
+
+/*
+ * Class:     ai_onnxruntime_OnnxSparseTensor
+ * Method:    getIndexBuffer
+ * Signature: (JJ)Ljava/nio/ByteBuffer;
+ */
+JNIEXPORT jobject JNICALL Java_ai_onnxruntime_OnnxSparseTensor_getIndexBuffer
+  (JNIEnv * jniEnv, jobject jobj, jlong apiHandle, jlong handle) {
+    (void) jobj; // Required JNI parameter not needed by functions which don't need to access their host object.
+    const OrtApi* api = (const OrtApi*) apiHandle;
+    OrtSparseFormat format;
+    checkOrtStatus(jniEnv,api,api->GetSparseTensorFormat((OrtValue*) handle, &format));
+    enum OrtSparseIndicesFormat indicesFormat;
+    switch (format) {
+        case ORT_SPARSE_COO:
+            indicesFormat = ORT_SPARSE_COO_INDICES;
+            break;
+        case ORT_SPARSE_CSRC:
+            indicesFormat = ORT_SPARSE_CSR_OUTER_INDICES;
+            break;
+        case ORT_SPARSE_BLOCK_SPARSE:
+            indicesFormat = ORT_SPARSE_BLOCK_SPARSE_INDICES;
+            break;
+        case ORT_SPARSE_UNDEFINED: {
+            throwOrtException(jniEnv,convertErrorCode(ORT_NOT_IMPLEMENTED),"Sparse format is ORT_SPARSE_UNDEFINED, cannot get indices");
+            return NULL;
+        }
+    }
+
+    OrtTensorTypeAndShapeInfo* info;
+    checkOrtStatus(jniEnv,api,api->GetSparseTensorIndicesTypeShape((OrtValue*) handle, indicesFormat, &info));
```

Comment:
>checkOrtStatus [](http://example.com/codeflow?start=4&length=14)

Please, remind how checkOrtStatus works. It does record Java exception, but otherwise is happily continuing calling native API. Should this some how stop on error since it may be feeding incorrect values?

Reply:
All the JNI code needs revising as the exception isn't necessarily propagated until the next call back into the JVM. So `checkOrtStatus` needs to be modified to return something that can be checked and all the places where it is called need to be modified with extra control flow to allow an early return, potentially several layers deep. I've not done that in this PR as it's going to be a large and confusing change which I'd like to keep separate from this one.

Reply:
I think we are going to SEGSEV if OrtTensorTypeAndShapeInfo is some stray pointer that was not properly populated because something errored out. The only thing we can do is to check for nullptr which we do not do since the pre-req is a valid instance. And if it is not nullptr then we can only assume it is valid.

Reply:
Ok, I'll work on a separate PR for the native fixes.